tests/fields/test_enum_field.py

```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+from enum import Enum
+
+import pytest
+
+from mongoengine import *
+from tests.utils import MongoDBTestCase, get_as_pymongo
+
+
+class Status(Enum):
+    NEW = 'new'
+    DONE = 'done'
+    
+    
+class ModelWithEnum(Document):
+    status = EnumField(Status)
+
+
+class Color(Enum):
+    RED = 1
+    BLUE = 2
+
+
+class ModelWithColor(Document):
+    color = EnumField(Color, default=Color.RED)
+
+
+class TestEnumField(MongoDBTestCase):
+    def test_storage(self):
+        model = ModelWithEnum(status=Status.NEW).save()
+        assert get_as_pymongo(model) == {"_id": model.id, "status": 'new'}
+
+    def test_set_enum(self):
+        ModelWithEnum.drop_collection()
+        m = ModelWithEnum(status=Status.NEW).save()
+        assert ModelWithEnum.objects(status=Status.NEW).count() == 1
+        assert ModelWithEnum.objects.first().status == Status.NEW
+        m.validate()
+
+    def test_set_by_value(self):
+        ModelWithEnum.drop_collection()
+        ModelWithEnum(status='new').save()
+        assert ModelWithEnum.objects.first().status == Status.NEW
+
+    def test_filter(self):
+        ModelWithEnum.drop_collection()
+        ModelWithEnum(status='new').save()
+        assert ModelWithEnum.objects(status='new').count() == 1
+        assert ModelWithEnum.objects(status=Status.NEW).count() == 1
+        assert ModelWithEnum.objects(status=Status.DONE).count() == 0
+
+    def test_change_value(self):
+        m = ModelWithEnum(status='new')
+        m.status = Status.DONE
+        m.validate()
+        assert m.status == Status.DONE
+
+    def test_set_default(self):
+        class ModelWithDefault(Document):
+            status = EnumField(Status, default=Status.DONE)
+
+        m = ModelWithDefault()
+        m.validate()
+        m.save()
+        assert m.status == Status.DONE
+
+    def test_enum_with_int(self):
+        m = ModelWithColor()
```

Comment:
same remark here, use
`m = ModelWithColor().save()` and remove the lines that call validate & save explicitly